test/blueprint/keyframe.spec.ts

```diff
@@ -0,0 +1,479 @@
+				{ slug: 'jellyfish-action-server', as: 'action-server' },
+				{ slug: 'jellyfish-tick-server', as: 'tick-server' },
+				{ slug: 'jellyfish-livechat', as: 'livechat' },
+				{ slug: 'redis' },
+				{ slug: 'postgres-db' },
+			],
+
+			output: {
+				type: 'keyframe',
+				slug: 'productOs-keyframe',
+
+				data: {
+					'behaviour-config': [
+						// Katapult k8s adapter will use this config for replicas count.
+						// Name convention: {instance-name}.{config-key-suffix}
+						{ name: 'api.replicas', type: 'int' },
+						// ...
+					],
+
+					connections: {
+						// Key: instance name ("as" in "requires").
+						api: {
+							// Key: name from contract requirement (as or slug).
+							// Value: instance name.
+							redis: 'redis',
+							'postgres-db': 'postgres-db',
+						},
+						ui: {
+							'jellyfish-api': 'api',
+						},
+						'action-server': {
+							redis: 'redis',
+							'postgres-db': 'postgres-db',
+						},
+						'tick-server': {
+							redis: 'redis',
+							'postgres-db': 'postgres-db',
+						},
+						livechat: {
+							'jellyfish-api': 'api',
+						},
+					},
+				},
+			},
+		},
+	};
+
+	const keyframe = contrato(productOsBlueprint, universe);
+
+	it('provides a valid keyframe', () => {
+		expect(keyframe).to.deep.equal({
+			type: 'keyframe',
+			slug: 'productOs-keyframe',
+			version: 'TODO',
+
+			// From selector in the blueprint. Also, defines the instance names.
+			requires: [
+				{
+					type: 'sw.containerized-service',
+					slug: 'jellyfish-api',
+					version: 'TODO',
+					as: 'api',
+				},
+				{
+					type: 'sw.containerized-service',
+					slug: 'jellyfish-ui',
+					version: 'TODO',
+					as: 'ui',
+				},
+				{
+					type: 'sw.containerized-worker',
+					slug: 'jellyfish-action-server',
+					version: 'TODO',
+					as: 'action-server',
+				},
+				{
+					type: 'sw.containerized-worker',
+					slug: 'jellyfish-tick-server',
+					version: 'TODO',
+					as: 'tick-server',
+				},
+				{
+					type: 'sw.containerized-service',
+					slug: 'jellyfish-livechat',
+					version: 'TODO',
+					as: 'livechat',
+				},
+				{ type: 'sw.database', slug: 'redis', version: 'TODO', as: 'redis' },
+				{
+					type: 'sw.database',
+					slug: 'postgres-db',
+					version: 'TODO',
+					as: 'postgres-db',
+				},
+			],
+
+			data: {
+				connections: productOsBlueprint.data.output.data.connections,
+			},
+		});
+	});
+
+	describe('k8s specs', () => {
+		const configStore = new YamlConfigStore({
+			yamlFile: { path: 'example-config.yaml' },
+		});
+		const k8sSpecs = katapult(keyframe, configStore);
+
+		it('have correct deployment definitions', () => {
+			expect(k8sSpecs.deployments).to.have.length(5);
+
+			// Check jellyfish-api deployment.
+			expect(k8sSpecs.deployments[0]).to.deep.equal({
+				apiVersion: 'app/v1', // TODO: What is a good value here for us?
+				kind: 'Deployment', // Per component type (sw.containerized-service or sw.containerized-worker).
```

Comment:
@ab77 @cmfcruz This is an example of k8s specs we'll get generated by katapult.
Please review. What are we missing?